MekHQ/src/mekhq/campaign/mission/AtBDynamicScenarioFactory.java

```diff
@@ -1945,18 +1948,24 @@ private static void setDeploymentTurns(AtBDynamicScenario scenario) {
      * ARRIVAL_TURN_STAGGERED_BY_LANCE is not implemented.
      * ARRIVAL_TURN_STAGGERED is processed just prior to scenario start instead (?)
      */
-    public static void setDeploymentTurns(BotForce botForce, int deployRound,
+    public static void setDeploymentTurns(BotForce botForce, ScenarioForceTemplate forceTemplate,
             AtBDynamicScenario scenario) {
         // deployment turns don't matter for transported entities
         List<Entity> untransportedEntities = scenario.filterUntransportedUnits(botForce.getEntityList());
         
-        if (deployRound == ScenarioForceTemplate.ARRIVAL_TURN_STAGGERED_BY_LANCE) {
+        if (forceTemplate.getArrivalTurn() == ScenarioForceTemplate.ARRIVAL_TURN_STAGGERED_BY_LANCE) {
             setDeploymentTurnsStaggeredByLance(untransportedEntities);
-        } else if (deployRound == ScenarioForceTemplate.ARRIVAL_TURN_AS_REINFORCEMENTS) {
-            setDeploymentTurnsForReinforcements(untransportedEntities, 0);
+        } else if (forceTemplate.getArrivalTurn() == ScenarioForceTemplate.ARRIVAL_TURN_AS_REINFORCEMENTS) {
+            if (forceTemplate.getForceAlignment() == ForceAlignment.Opposing.ordinal()) {
+                setDeploymentTurnsForReinforcements(untransportedEntities, scenario.getHostileReinforcementDelayReduction());
+            } else if (forceTemplate.getForceAlignment() != ForceAlignment.Third.ordinal()) {
+                setDeploymentTurnsForReinforcements(untransportedEntities, scenario.getFriendlyReinforcementDelayReduction());
+            } else {
+                setDeploymentTurnsForReinforcements(untransportedEntities, 0);
+            }
```

Comment:
Switch-case would be a better format, and the ordinal isn't required.

Reply:
It's required, otherwise off-by-one errors abound. I regret the choice to use the ordinal instead of explicit values, but have no desire to get into it at this time.

Reply:
Ah... leave it then